src/processing/sobel.rs

```diff
@@ -13,15 +13,18 @@ where
     /// Type to output
     type Output;
     /// Returns the magnitude output of the sobel - an image of only lines
-    fn apply_sobel(&self) -> Result<Self, Error>;
+    fn apply_sobel(&self) -> Result<Self::Output, Error>;
 
     /// Returns the magntitude and rotation outputs for use in other algorithms
     /// like the Canny edge detector. Rotation is in radians
     fn full_sobel(&self) -> Result<(Self::Output, Self::Output), Error>;
 }
 
-fn get_edge_images<T>(mat: &Array3<T>) -> Result<(Array3<T>, Array3<T>), Error>
+fn get_edge_images<T, U>(
+    mat: &ArrayBase<U, Ix3>,
+) -> Result<(ArrayBase<OwnedRepr<T>, Ix3>, ArrayBase<OwnedRepr<T>, Ix3>), Error>
```

Comment:
`Result<(Array3<T>, Array3<T>), Error>` would probably be more idiomatic (using the aliases where possible).